### DISCUSSION TITLE
Error: spawn yarn ENOENT

### DISCUSSION THREAD

**Discussion by philippedeforge** [eclipse-theia/theia#14494] [General]
Dear all, I would like to discover theia but I'm facing this issue. yarn is installed globally (version 1.22.22) and in the PATH but the issue is still there. Any Idea ? 

Thanks

C:\Users\philippe\my-theia-app>yo theia-extension
? The extension's type Hello World
? The extension's name Test
Test
identical package.json
identical lerna.json
identical .gitignore
identical README.md
identical .vscode\launch.json
identical Test\package.json
identical Test\tsconfig.json
identical Test\src\browser\Test-frontend-module.ts
identical Test\src\browser\Test-contribution.ts
identical Test\README.md
identical browser-app\package.json
identical electron-app\package.json
Installing dependencies
node:events:495
      throw er; // Unhandled 'error' event
      ^

Error: spawn yarn ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:284:19)
    at onErrorNT (node:internal/child_process:477:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:290:12)
    at onErrorNT (node:internal/child_process:477:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn yarn',
  path: 'yarn',
  spawnargs: []
}

Node.js v18.20.5

**Comment by JonasHelming** [comment]
It indeed looks like yarn is not installed correctly. 
`yarn --version`
In the same terminal works?

**Comment by philippedeforge** [comment]
Hi Jonas,

Yes, it works:

C:\Users\philippe\my-theia-app>yarn --version
1.22.22

**Comment by JonasHelming** [comment]
Can node generally spawn yarn?

node -e "require('child_process').spawnSync('yarn', ['--version'], { stdio: 'inherit' })"

If this fails with ENOENT, it confirms Node.js cannot find Yarn, even though it's in the PATH.

**Comment by philippedeforge** [comment]
I've run the following command : 

>node -e "require('child_process').exec('yarn --version', (err, stdout, stderr) => { if (err) { console.error('Yarn not found'); } else { console.log(stdout); } })"
1.22.22

The result confirm the yarn version.

**Comment by JonasHelming** [comment]
OK, can you run in a fresh directory:
yo theia-extension --skip-install
and then
yarn install?

**Reply by philippedeforge** [reply]
There is still one error but it seems that your workaround works.

C:\Users\philippe\theiatesting>yo theia-extension --skip-install
? The extension's type Hello World
? The extension's name Test
Test
   create package.json
   create lerna.json
   create .gitignore
   create README.md
   create .vscode\launch.json
   create Test\package.json
   create Test\tsconfig.json
   create Test\src\browser\Test-frontend-module.ts
   create Test\src\browser\Test-contribution.ts
   create Test\README.md
   create browser-app\package.json
   create electron-app\package.json

C:\Users\frq08515\theiatesting>yarn install
yarn install v1.22.22
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
info There appears to be trouble with your network connection. Retrying...
warning workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > browser-app > @theia/terminal > xterm-addon-fit@0.8.0: This package is now deprecated. Move to @xterm/addon-fit instead.
warning workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > browser-app > @theia/terminal > xterm-addon-search@0.13.0: This package is now deprecated. Move to @xterm/addon-search instead.
warning workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > browser-app > @theia/terminal > xterm@5.3.0: This package is now deprecated. Move to @xterm/xterm instead.
info There appears to be trouble with your network connection. Retrying...
[3/5] Fetching packages...
info There appears to be trouble with your network connection. Retrying...
[4/5] Linking dependencies...
warning "workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > browser-app > @theia/messages > react-perfect-scrollbar@1.5.8" has unmet peer dependency "react@>=16.3.3".
warning "workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > browser-app > @theia/messages > react-perfect-scrollbar@1.5.8" has unmet peer dependency "react-dom@>=16.3.3".
warning "workspace-aggregator-d18d93d5-4a85-468f-bab4-fc4156f7799b > Test > @theia/core > inversify > @inversifyjs/core > @inversifyjs/reflect-metadata-utils@0.2.3" has incorrect peer dependency "reflect-metadata@0.2.2".
[5/5] Building fresh packages...
[1/12] ⠁ command-join
[2/12] ⠂ keytar
[8/12] ⠂ puppeteer
[4/12] ⠂ drivelist
error C:\Users\philippe\theiatesting\node_modules\puppeteer: Command failed.
Exit code: 1
Command: node install.mjs
Arguments:
Directory: C:\Users\philippe\theiatesting\node_modules\puppeteer
Output:
**INFO** Skipping Firefox download as instructed.
Error: ERROR: Failed to set up chrome-headless-shell v127.0.6533.119! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download.
    at downloadBrowser (file:///C:/Users/philippe/theiatesting/node_modules/puppeteer/lib/esm/puppeteer/node/install.js:26:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 1)
    at async downloadBrowsers (file:///C:/Users/philippe/theiatesting/node_modules/puppeteer/lib/esm/puppeteer/node/install.js:84:9) {
  [cause]: Error: Got status code 407
      at file:///C:/Users/philippe/theiatesting/node_modules/@puppeteer/browsers/lib/esm/httpUtil.js:105:31
      at ClientRequest.requestCallback (file:///C:/Users/philippe/theiatesting/node_modules/@puppeteer/browsers/lib/esm/httpUtil.js:45:13)
      at Object.onceWrapper (node:events:632:26)
      at ClientRequest.emit (node:events:517:28)
      at HTTPParser.parserOnIncomingClient (node:_http_client:700:27)
      at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
      at Socket.socketOnData (node:_http_client:541:22)
      at Socket.emit (node:events:517:28)

**Reply by lthomas122** [reply]
```
[cause]: Error: Got status code 407
```
This is a proxy authentication error. Are you using a VPN or an enterprise network that utilises a proxy server?
You may need to configure [Yarn proxy settings](https://documentation.coremedia.com/cmcc-10/current/webhelp/frontend-en/content/InstallationProxy.html). 

If that's not the case, try clearing cache with `yarn cache clean`
